Source/SwiftLintFramework/Rules/RuleConfigs/SeverityLevelsConfig.swift

```diff
@@ -10,24 +10,25 @@ import Foundation
 
 public struct SeverityLevelsConfig: RuleConfig, Equatable {
     var warning: Int
-    var error: Int
+    var error: Int?
 
     var params: [RuleParameter<Int>] {
-        return [RuleParameter(severity: .Error, value: error),
+        if let error = error {
+            return [RuleParameter(severity: .Error, value: error),
                 RuleParameter(severity: .Warning, value: warning)]
+        }
+        return [RuleParameter(severity: .Warning, value: warning)]
     }
 
     mutating public func setConfig(config: AnyObject) throws {
         if let config = [Int].arrayOf(config) where !config.isEmpty {
             warning = config[0]
-            if config.count > 1 {
-                error = config[1]
-            }
+            error = (config.count > 1) ? config[1] : nil
         } else if let config = config as? [String: AnyObject] {
             if let warningNumber = config["warning"] as? Int {
                 warning = warningNumber
-            }
-            if let errorNumber = config["error"] as? Int {
+                error = config["error"] as? Int
+            } else if let errorNumber = config["error"] as? Int {
                 error = errorNumber
```

Comment:
I thought about it, but it would change the behavior of the method in cases of config:

- being an empty dictionary
- containing keys besides 'warning' and 'error' (because of typos, for example)

The first one can be addressed easily enough with a where clause. The second case might be ok too, since we're documenting this as a breaking change. However, since there's no formal config validator, I wasn't quite comfortable going that route.

What do you think?